MIPTools.def

```diff
@@ -392,15 +396,91 @@ From: amd64/ubuntu:20.04
+    set -e
+
+    # Set default values for paths
+    output_path="/opt/analysis"
+    config_path="/opt/resources/basespace.cfg"
+
+    help() {
+        echo "Download data from the Illumina BaseSpace Sequence Hub."
+        echo ""
+        echo "Usage:"
+        echo "  singularity run [options] --app bs_download <container>"\
+        "[app_options]"
+        echo ""
+        echo "Options:"
+        echo "  See 'singularity run'."
+        echo ""
+        echo "App Options:"
+        echo "  -i    Required. The run ID of the data to download."
+        echo "  -o    The path to the output directory."
+        echo "        Default: '/opt/analysis'."
+        echo "  -c    The path to the authentication credentials file."
+        echo "        This file is created by 'bs auth'. For additional"
+        echo "        information see the help page for that command."
+        echo "        Default: '/opt/resources/basespace.cfg'."
+        echo "  -h    Print the help page."
+        echo ""
+        echo "Examples:"
+        echo "  # Set paths"
+        echo "  $ resource_dir=/bin/MIPTools/base_resources"
+        echo "  $ run_dir=/work/usr/example"
+        echo ""
+        echo "  # Run app"
+        echo "  $ singularity run \\"
+        echo "    -B \$resource_dir:/opt/resources \\"
+        echo "    -B \$run_dir:/opt/analysis \\"
+        echo "    --app bs_download <container> -i <run_id>"
+    }
+    
+    # Parse options
+    while getopts "i:o:c:h" opt; do
+      case "${opt}" in
+        c) config_path=${OPTARG} ;;
+        h) help
+           exit 1 ;;
+        i) run_id=${OPTARG} ;;
+        o) output_path=${OPTARG} ;;
+        *) help
+           exit 1 ;;
+      esac
+    done
+    
+    # Ensure run_id is specified
+    if [ ! $run_id ]; then
+      echo "Argument -i must be provided"
+      echo "$usage" >&2
+      exit 1
+    fi
+
+    # Read data from config file
+    # Remove whitespace from each line and export each line as a variable
+    export BASESPACE_API_SERVER=$(sed "1q;d" ${config_path} | sed "s/.*=.//g")
+    export BASESPACE_ACCESS_TOKEN=$(sed "2q;d" ${config_path} | sed "s/.*=.//g")
+
+    # Download data
+    bs download run -i ${run_id} -o ${output_path}/${run_id}
+
+#################################################################
+##                   Superseded Download App                   ##
+#################################################################
+%apprun download_superseded
     # Exit if something fails or if have unset object
     set -eu
 
     help() {
         echo "Download data from the Illumina BaseSpace Sequence Hub."
         echo ""
+        echo "Superseded Note:"
+        echo "  Please note that this app has been superseded by the download" 
+        echo "  app, which uses the basespace command line interface for" 
+        echo "  downloading data."
+        echo ""
         echo "Usage:"
         echo "  singularity run [options] --app download <container>"\
```

Comment:
```suggestion
        echo "  singularity run [options] --app download_superseeded <container>"\
```

Reply:
I think it is probably a good idea to build a container with the current branch code and make sure it runs before merging this. I don't think it would work as it is now.